src/Http/ServerRequest.php

```diff
@@ -466,18 +471,35 @@ public function referer(bool $local = true): ?string
      *
      * @param string $name The method called
      * @param array $params Array of parameters for the method call
-     * @return bool
+     * @return mixed
      * @throws \BadMethodCallException when an invalid method is called.
      */
-    public function __call(string $name, array $params): bool
+    public function __call(string $name, array $params): mixed
     {
         if (str_starts_with($name, 'is')) {
             $type = strtolower(substr($name, 2));
 
             array_unshift($params, $type);
 
             return $this->is(...$params);
+        } elseif (str_starts_with($name, 'get')) {
+            // e.g. getCookieAsInt(), getQueryAsBool()
+            $parts = explode('As', $name);
+            if (count($parts) !== 2) {
+                return $this->$name(...$params);
```

Comment:
Should this be an error? If the method starts with `get`, contains `As` and has more than two `As` the method is likely not to callable and we should error.

Reply:
I reworked it just a tad so that it will only do the new type coercion logic if the number of parts were exploded == 2. I don't love it, but it felt better than copying the `BadRequestException` throw that is on the final lines of the method. Felt like it would be better to just fall through and let the exception throw? Advice welcome.